### ISSUE TITLE
[Feature Request] SoftargMax function

### ISSUE THREAD

**Issue by Naman-ntc** [pytorch/pytorch#7739]
Softargmax is used quite many place. It can be used as a differentiable alternative to argmax function and thus comes handy! [A brief idea](https://bouthilx.wordpress.com/2013/04/21/a-soft-argmax/) about softargmax function.

What do you think about adding it into the nn Module?

I used `nn.Softmax` as well as `torch.arange` to implement it [here on forums](https://discuss.pytorch.org/t/softargmax-in-pytorch-in-2d/17931).

I can create a PR if you just mention relevant files I need to play around for the function.
